replace stack with nix, bump to ghc 8.8.4

michaelglass opened the pull request:
bumps ghc to 8.8.4, replaces stack with nix

michaelglass committed:
replace stack with nix, bump to ghc 8.8.4

michaelglass committed:
stop testing ghc 8.6.5

jwoudenberg reviewed (approved):
Looks great. We should look into failing CI though. Will do later!

omnibs committed:
Bump 21.05 ghc

omnibs committed:
Blindly follow instructions from the internet

Trying these out:
  https://kodimensional.dev/posts/2019-02-25-haskell-travis

omnibs committed:
Update cabal? Nix uses 3.2

omnibs committed:
Try haskell-ci cli app's config...

Can't build for macos for some reason, but I guess we don't expect
cross-compilation errors in this kinda library?

omnibs committed:
Make cabal info pass

omnibs committed:
Add doctests and osx jobs

omnibs committed:
Relax deps for nixpkgs-21.05

omnibs committed:
Add nixpkgs-21.05 and build the lib on it too

omnibs commented:
I'm not super fond of this setup for development. The builds don't seem to be getting cached. I change a .nix file and get a full rebuild of the library. might be cabal2nix is thinking every file under this repo is a dependency? Not sure.

That said, we change this so infrequently that I think it doesn't merit the effort of getting to an awesome place with dev tooling. 🤷 

The builds are now working for Linux. OSX jobs are taking a while, but I don't expect them to fail.
